src/framework/components/rigid-body/system.js

```diff
@@ -705,6 +690,42 @@ Object.assign(pc, function () {
             this.contactPointPool.freeAll();
             this.contactResultPool.freeAll();
             this.singleContactResultPool.freeAll();
+        },
+
+        onUpdate: function (dt) {
+            // #ifdef PROFILER
+            this._stats.physicsStart = pc.now();
+            // #endif
+
+            // Check to see whether we need to update gravity on the dynamics world
+            var gravity = this.dynamicsWorld.getGravity();
+            if (gravity.x() !== this.gravity.x || gravity.y() !== this.gravity.y || gravity.z() !== this.gravity.z) {
+                gravity.setValue(this.gravity.x, this.gravity.y, this.gravity.z);
+                this.dynamicsWorld.setGravity(gravity);
+            }
+
+            // Update the transforms of all bodies
+            this.dynamicsWorld.stepSimulation(dt, this.maxSubSteps, this.fixedTimeStep);
+
+            // Update the transforms of all entities referencing a body
+            var components = this.store;
+            for (var id in components) {
+                if (components.hasOwnProperty(id)) {
+                    var entity = components[id].entity;
+                    var componentData = components[id].data;
+                    if (componentData.body && componentData.body.isActive() && componentData.enabled && entity.enabled) {
+                        if (componentData.type === pc.BODYTYPE_DYNAMIC) {
+                            entity.rigidbody.syncBodyToEntity();
+                        } else if (componentData.type === pc.BODYTYPE_KINEMATIC) {
+                            entity.rigidbody._updateKinematic(dt);
+                        }
+                    }
+
+                }
+            }
+
+            if (!this.dynamicsWorld.setInternalTickCallback)
+                this._checkForCollisions();
```

Comment:
Personally, I would pass `this.dynamicsWorld` and `dt` here, since that's the function signature now.

Reply:
OK, forget this since `this.dynamicWorld` does not equal the `number` value passed from ammo.js.